30413/Zarnescu Radu/assignment3/main.c

```diff
@@ -0,0 +1,154 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+typedef struct node
+{
+
+    char data;
+    struct node * next, * previous;
+
+}NodeL;
+
+NodeL * first, * last;
+
+typedef struct node_type
+{
+
+    char id;
+    struct node_type * left, * right;
+
+}NodeT;
+
+void initializeDLL()
+{
+
+    first = NULL;
+    last = NULL;
+
+}
+
+void addLast(char c)
+{
+
+    if(first == NULL)
+    {
+        first = (NodeL*)malloc(sizeof(NodeL));
+        first->data = c;
+        first->next = NULL;
+        first->previous = NULL;
+        last = first;
+    }
+    else
+    {
+        NodeL * newElement = (NodeL*)malloc(sizeof(NodeL));
+        newElement->data = c;
+        newElement->previous = last;
+        last->next = newElement;
+        newElement->next = NULL;
+        last = newElement;
+    }
+
+}
+
+NodeT * createBinTree()
+{
+    NodeT * p = (NodeT*)malloc(sizeof(NodeT));
+    char c;
+
+    scanf("%c", &c);
+    if(c == '*')
+    {
+        return NULL;
+    }
+    else
+    {
+        if(p == NULL)
+        {
+            printf("out of space in binary tree\n");
+        }
+        p->id = c;
+        p->left = createBinTree();
+        p->right = createBinTree();
+    }
+    return p;
+}
+
+void getListFromTree(NodeT * root)
+{
+
+    if(root == NULL)
+    {
+        addLast('*');
+    }
+    else
+    {
+        addLast(root->id);
+        getListFromTree(root->left);
+        getListFromTree(root->right);
+    }
+
+}
+
+NodeL * current;
+
+NodeT * getTreeFromList()
```

Comment:
it would've been better that those 2 methods (tree to list and list to tree) expected the tree/list as parameter and return the list/tree they're building out of it.. somehting like `NodeT *getTreeFromList(NodeT *list)`